Source/santactl/Commands/SNTCommandFileInfo.m

```diff
@@ -268,30 +320,30 @@ - (SNTAttributeBlock)codeSigned {
 }
 
 - (SNTAttributeBlock)rule {
-  return ^id (SNTCommandFileInfo *fi) {
-    __block SNTEventState s;
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
-      [fi.daemonConn resume];
-    });
+  return ^id (SNTCommandFileInfo *cmd, SNTFileInfo *fileInfo) {
+    // If we previously were unable to connect, don't try again.
+    if (cmd.daemonUnavailable) return @"Could not communicate with daemon";
+    static dispatch_once_t token;
+    dispatch_once(&token, ^{ [cmd.daemonConn resume]; });
+    __block SNTEventState state;
     dispatch_semaphore_t sema = dispatch_semaphore_create(0);
-    if (!fi.csc) {
-      NSError *error;
-      fi.csc = [[MOLCodesignChecker alloc] initWithBinaryPath:fi.path(fi) error:&error];
-    }
-    [[fi.daemonConn remoteObjectProxy] decisionForFilePath:fi.path(fi)
-                                                fileSHA256:fi.propertyMap[kSHA256](fi)
-                                         certificateSHA256:fi.csc.leafCertificate.SHA256
-                                                     reply:^(SNTEventState state) {
-      if (state) s = state;
+    NSError *error;
+    MOLCodesignChecker *csc = [[MOLCodesignChecker alloc] initWithBinaryPath:fileInfo.path
+                                                                       error:&error];
+    [[self.daemonConn remoteObjectProxy] decisionForFilePath:fileInfo.path
+                                                  fileSHA256:fileInfo.SHA256
+                                           certificateSHA256:csc.leafCertificate.SHA256
+                                                       reply:^(SNTEventState s) {
+      state = s;
       dispatch_semaphore_signal(sema);
     }];
     if (dispatch_semaphore_wait(sema, dispatch_time(DISPATCH_TIME_NOW, 5 * NSEC_PER_SEC))) {
-      return @"Cannot communicate with daemon";
+      cmd.daemonUnavailable = YES;
+      return @"Could not communicate with daemon";
```

Comment:
nit: Using the string "Could not communicate with daemon" twice. Consider making this a constant or at least a local variable.